capstone-backend/src/main/java/com/google/sps/servlets/MeetingServlet.java

```diff
@@ -66,7 +67,34 @@ public Calendar createCalendar(JsonObject jsonObject) {
     return service;
   }
 
+  private JsonObject extractTokenFromFirestore(String userID) {
 
+    JsonObject tokenObject = new JsonObject();
+
+    try {
+
+      DocumentSnapshot document = db.collection("users").document(userID).get().get();
+
+      Map<String, Object> user = document.getData();
+
+      // Delete '{}'
+      StringBuilder tokenString = new StringBuilder(user.get("tokenObj").toString());
+      tokenString.deleteCharAt(0);
+      tokenString.deleteCharAt(tokenString.length() - 1);
+
+      Map<String, String> tokenMap = Arrays.stream(tokenString.toString().split(",")).map(s -> s.split("="))
+          .collect(Collectors.toMap(s -> s[0].replaceAll("\\s", ""), s -> s[1]));
+
+      for (String key : tokenMap.keySet()) {
+        tokenObject.addProperty(key, tokenMap.get(key));
+      }
```

Comment:
Why are you parsing it like this instead of Gson parse on the tokenObj? It looks like you are just manually parsing the Gson instead of the library

Reply:
Hi Antonio, I initially tried to parse it using ```gson``` but the way Firestore sends and formats the user's ```tokenObj``` is incompatible with ```Json.parser```.